File: src/main/java/org/dukcode/ps/codetree/trail02/chapter04/lesson02/IntroConvertToBinary.java
package org.dukcode.ps.codetree.trail02.chapter04.lesson02;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;

/**
 * @see <a
 * href="https://www.codetree.ai/ko/trails/complete/curated-cards/intro-convert-to-binary/description">LINK</a>
 */
public class IntroConvertToBinary {

  private static BufferedReader br;
  private static BufferedWriter bw;

  public static void main(String[] args) throws IOException {
    br = new BufferedReader(new InputStreamReader(System.in));
    bw = new BufferedWriter(new OutputStreamWriter(System.out));

    int decimal = Integer.parseInt(br.readLine());
    bw.write(toBinary(decimal));

    br.close();
    bw.close();

  }

  private static String toBinary(int decimal) {
    if (decimal == 0) {
      return "0";
    }

    StringBuilder sb = new StringBuilder();

    while (decimal > 0) {
      sb.append(decimal % 2);
      decimal /= 2;
    }

    return sb.reverse().toString();
  }
}

File: src/main/kotlin/org/dukcode/ps/codetree/trail02/chapter04/lesson02/IntroConvertToBinary.kt
package org.dukcode.ps.codetree.trail02.chapter04.lesson02.introConvertToBinary

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

private val br = BufferedReader(InputStreamReader(System.`in`))
private val bw = BufferedWriter(OutputStreamWriter(System.out))

/**
 * @see <a href="https://www.codetree.ai/ko/trails/complete/curated-cards/intro-convert-to-binary/description">LINK</a>
 */
fun main() {
    val n = br.readLine().toInt()
    val binary = n.toBinaryString()

    bw.write(binary)

    bw.flush()
}

private fun Int.toBinaryString(): String {
    if (this == 0) {
        return "0"
    }

    var value = this
    val digits =
        buildString {
            while (value > 0) {
                append(value % 2)
                value /= 2
            }
        }

    return digits.reversed()
}
